### ISSUE TITLE
Can't push to git repository

### ISSUE THREAD

**Issue by anaspeople** [jbox-web/redmine_git_hosting#147]
I have successfully installed Redmine 2.4.2 and the latest versions of the plugins redmine_git_hosting and redmine_plugin_views_revisions in ubuntu 12.04.4
I have followed the steps in your wiki and user www-data connects correctly to gitolite.

When I go to repository tab, I can't go past the message "Repository is empty. Get started by following the instructions below."
I have tried following the instructions here but when I try git push -u origin master I get the following error:

```
fatal: Authentication failed for 'https://user@projects/net/net.git/'
```

It does not accept my redmine user's password 
I also activated autocreation of repositories for new projects. The name for the previous test project is "net".

Looking into the logs I found that it looks like it is not detecting my user but anonymous instead:

```
2014-03-13 17:41:51 +0100 INFO [GitSmartHttp] ##########################
2014-03-13 17:42:05 +0100 INFO [GitSmartHttp] ["  Current user: anonymous"]
2014-03-13 17:42:05 +0100 INFO [GitSmartHttp] ###### AUTHENTICATION ######
2014-03-13 17:42:05 +0100 INFO [GitSmartHttp] git_params      : info, refs
2014-03-13 17:42:05 +0100 INFO [GitSmartHttp] repo_path       : net/net.git
2014-03-13 17:42:05 +0100 INFO [GitSmartHttp] is_push         : true
2014-03-13 17:42:05 +0100 INFO [GitSmartHttp] project name    : net
2014-03-13 17:42:05 +0100 INFO [GitSmartHttp] public project  : true
2014-03-13 17:42:05 +0100 INFO [GitSmartHttp] repository name : net
2014-03-13 17:42:05 +0100 INFO [GitSmartHttp] repository path : repositories/net.git
2014-03-13 17:42:05 +0100 INFO [GitSmartHttp] Valid push
2014-03-13 17:42:05 +0100 INFO [GitSmartHttp] ##########################
```

I found this in the wiki but don't know how to activate it:

```
Also note that you will need to ensure that Basic Auth headers are being passed properly to Rails for this to work properly. In Apache with mod_fcgid this may mean you need to add "Passheader Authorization" into the virtual host configuration file.
```

Note that I am new to the git world and I don't know if I should set deployment keys / user keys, although I already tried adding my public key to both settings without any success.

Basically I'm interested on making this work through https.
